pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -128,19 +128,17 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 		bestHint = defaultHint
 	}
 
-	// topology manager returns the hint that does not satisfy completely the container request
-	// we should extend this hint to the one who will satisfy the request and include the current hint
-	if !isAffinitySatisfyRequest(machineState, bestHint.NUMANodeAffinity, requestedResources) {
```

Comment:
can you please add some explanation about why its safe to remove this conditional check and apply the extendedHint in all conditions?

Reply:
@swatisehgal @ffromani This specific code removal, can fix https://github.com/kubernetes/kubernetes/issues/120733

Reply:
can you please add some explanation about why removing these code can help with fixing https://github.com/kubernetes/kubernetes/issues/120733 ?
It will still call `extendTopologyManagerHint`, and return error

Reply:
@AllenXu93 It's quite cumbersome, because it requires to understand the root cause for #120733.

In brief the general idea is when `extendTopologyManagerHint` is running it makes sure the given `TopologyHint` follows the architectural allocation decision dictated by MemoryManager, for example: https://kubernetes.io/blog/2021/08/11/kubernetes-1-22-feature-memory-manager-moves-to-beta/#single-vs-cross-numa-node-allocation

However, I suggested a different resolution for #120733 - #123443 which you're welcome to review and which explain the problem in details.